src/main/java/org/mitre/synthea/export/FhirStu3.java

```diff
@@ -810,6 +820,313 @@ private static BundleEntryComponent encounterClaim(BundleEntryComponent personEn
+        .setValue("93480754268");
+
+    eob.setStatus(org.hl7.fhir.dstu3.model.ExplanationOfBenefit.ExplanationOfBenefitStatus.ACTIVE);
+    eob.setCreated(encounterResource.getPeriod().getEnd());
+    eob.setType(claim.getType());
+
+    List<ExplanationOfBenefit.DiagnosisComponent> eobDiag = new ArrayList<>();
+
+    for (org.hl7.fhir.dstu3.model.Claim.DiagnosisComponent claimDiagnosis : claim.getDiagnosis()) {
+      ExplanationOfBenefit.DiagnosisComponent diagnosisComponent =
+          new ExplanationOfBenefit.DiagnosisComponent();
+      diagnosisComponent.setDiagnosis(claimDiagnosis.getDiagnosis());
+      diagnosisComponent.getType().add(new CodeableConcept()
+          .addCoding(new Coding()
+              .setCode("principal")
+              .setDisplay("The single medical diagnosis that is most relevant "
+                  + "to the patient's chief complaint or need for treatment.")
+              .setSystem("https://bluebutton.cms.gov/resources/codesystem/diagnosis-type")));
+      diagnosisComponent.setSequence(claimDiagnosis.getSequence());
+      diagnosisComponent.setPackageCode(claimDiagnosis.getPackageCode());
+      eobDiag.add(diagnosisComponent);
+    }
+    eob.setDiagnosis(eobDiag);
+    List<ExplanationOfBenefit.ProcedureComponent> eobProc = new ArrayList<>();
+
+    // get the insurance info at the time that the encounter happened
+    String insurance = encounter.claim.insurance.getInsuranceName();
+    eob.setInsurance(new ExplanationOfBenefit.InsuranceComponent()
+        .setCoverageTarget(new Coverage()
+            .addPayor(new Reference()
+                .setReference(insurance))));
+
+
+    for (ProcedureComponent proc : claim.getProcedure()) {
+      ExplanationOfBenefit.ProcedureComponent p = new ExplanationOfBenefit.ProcedureComponent();
+      p.setDate(proc.getDate());
+      p.setSequence(proc.getSequence());
+      p.setProcedure(proc.getProcedure());
+    }
+
+    List<ExplanationOfBenefit.ItemComponent> eobItem = new ArrayList<>();
+
+    double totalPayment = 0;
+    // Get all the items info from the claim
+
+    for (ItemComponent item : claim.getItem()) {
+
+      ExplanationOfBenefit.ItemComponent itemComponent = new ExplanationOfBenefit.ItemComponent();
+
+      itemComponent.setSequence(item.getSequence());
+      itemComponent.setCareTeamLinkId(item.getCareTeamLinkId());
+      itemComponent.setDiagnosisLinkId(item.getDiagnosisLinkId());
+      itemComponent.setInformationLinkId(item.getInformationLinkId());
+      itemComponent.setNet(item.getNet());
+      itemComponent.setQuantity(item.getQuantity());
+      itemComponent.setUnitPrice(item.getUnitPrice());
+      itemComponent.setEncounter(item.getEncounter());
+
+      // can probably set code based on encounter type with switch statement
+      itemComponent.setService(new CodeableConcept().addCoding(new Coding()
+          .setSystem("https://bluebutton.cms.gov/resources/codesystem/hcpcs")
```

Comment:
In this for-loop, these ItemComponents align with Claim items. 

Therefore, if the item refers to a Procedure, then this should be the Procedure SNOMED code. If this item was a Diagnostic Report, then this item should be the proper LOINC code.

HCPCS is a superset of codes which includes CPT, which we do not support. This should not be hard-coded.